docs/csharp/programming-guide/concepts/async/async-return-types.md

```diff
@@ -1,63 +1,74 @@
-
-When `GetLeisureHours` is called from within an await expression in the `ShowTodaysInfo` method, the await expression retrieves the integer value (the value of `leisureHours`) that's stored in the task returned by the `GetLeisureHours` method. For more information about await expressions, see [await](../../../language-reference/operators/await.md).  
-  
-You can better understand how `await` retrieves the result from a `Task<T>` by separating the call to `GetLeisureHours` from the application of `await`, as the following code shows. A call to method `GetLeisureHours` that isn't immediately awaited returns a `Task<int>`, as you would expect from the declaration of the method. The task is assigned to the `integerTask` variable in the example. Because `integerTask` is a <xref:System.Threading.Tasks.Task%601>, it contains a <xref:System.Threading.Tasks.Task%601.Result> property of type `TResult`. In this case, `TResult` represents an integer type. When `await` is applied to `integerTask`, the await expression evaluates to the contents of the <xref:System.Threading.Tasks.Task%601.Result%2A> property of `integerTask`. The value is assigned to the `ret` variable.  
-  
-> [!IMPORTANT]
-> The <xref:System.Threading.Tasks.Task%601.Result%2A> property is a blocking property. If you try to access it before its task is finished, the thread that's currently active is blocked until the task completes and the value is available. In most cases, you should access the value by using `await` instead of accessing the property directly. <br/> The previous example retrieved the value of the <xref:System.Threading.Tasks.Task%601.Result%2A> property to block the main thread so that the `ShowTodaysInfo` method could finish execution before the application ended.  
+For more information about async methods, see [Asynchronous programming with async and await (C#)](./index.md).
+
+Several other types also exist that are specific to Windows workloads:
+
+- <xref:System.Windows.Threading.DispatcherOperation>, for async operations limited to Windows.
+- <xref:Windows.Foundation.IAsyncAction>, for async actions in UWP that do not return a value.
+- <xref:Windows.Foundation.IAsyncActionWithProgress%601>, for async actions in UWP that report progress but do not return a value.
+- <xref:Windows.Foundation.IAsyncOperation%601>, for async operations in UWP that return a value.
+- <xref:Windows.Foundation.IAsyncOperationWithProgress%602>, for async operations in UWP that report progress and return a value.
+
+## Task return type
+
+Async methods that don't contain a `return` statement or that contain a `return` statement that doesn't return an operand usually have a return type of <xref:System.Threading.Tasks.Task>. Such methods return `void` if they run synchronously. If you use a <xref:System.Threading.Tasks.Task> return type for an async method, a calling method can use an `await` operator to suspend the caller's completion until the called async method has finished.
```

Comment:
I've seen async methods that returns a `Task`, but they have `return Task.CompletedTask`.
This contradicts "Async methods that don't contain a `return` statement or that contain a `return` statement that doesn't return an operand".

Reply:
This is not a contradiction. The `Task.CompletedTask` is a `Task`.

Reply:
Yes, but the current wording says that a task-returning method has either a return without operand or no return at all. Unless I'm missing something, I'd say that it has a return without operand, a return with operand whose type is a Task, or no return at all.